rhel/Dockerfile

```diff
@@ -12,16 +33,22 @@ LABEL name="Kong" \
       summary="Next-Generation API Platform for Modern Architectures" \
       description="Next-Generation API Platform for Modern Architectures"
 
+COPY --from=DOWNLOADER /tmp/kong.rpm /tmp/kong.rpm
 COPY LICENSE /licenses/
 
-RUN yum install -y https://bintray.com/kong/kong-rpm/download_file?file_path=rhel/7/kong-$KONG_VERSION.rhel7.amd64.rpm && \
-    yum clean all && \
-    # OpenShift specific. OpenShift runs containers using an arbitrarily assigned user ID.
-    # This user doesn't have access to change file permissions during runtime, they have to be changed during image building.
-    # https://docs.okd.io/latest/creating_images/guidelines.html#use-uid
-    mkdir -p "/usr/local/kong" && \
-    chgrp -R 0 "/usr/local/kong" && \
-    chmod -R g=u "/usr/local/kong"
+RUN yum install -y -q unzip \
+	&& yum clean all -q \
+	&& rm -fr /var/cache/yum/* /tmp/yum_save*.yumtx /root/.pki
+
+RUN useradd kong \
+	&& mkdir -p "/usr/local/kong" \
+	&& yum install -y /tmp/kong.rpm \
+	&& yum clean all \
+	&& rm /tmp/kong.rpm \
+	&& chown -R kong:0 /usr/local/kong \
+	&& chmod -R g=u /usr/local/kong
+
+USER kong
```

Comment:
`USER kong` is needed because of OpenShift, right?

But having tools like curl and wget and the ability to run as root are often useful for development, and Kubernetes' lack of of a `--user` makes it difficult to use these images for testing. There are security reasons to not include tools like curl and wget and run as root but I wonder if we should include a Dockerfile.debug for development that has `USER root` and wget, curl, httpie, jq, etc. It's not a blocker for this PR but it would be useful for users wanting to use our Kong base image for development. If we don't include it, they would have to run as root and install packages in the running container. For Kubernetes it's extra work to modify build and push to the registry. However, I'm not sure if including a Dockerfile.debug is a common.

Reply:
Docker development files exist in a few places including kong-build-tools, pongo, and gojira. Let's leave unifying / making an "official" Kong docker development image as a future us task.

OpenShift had challenges with previous version of Kong because it was a hard requirement that it had to run as UID 1337.

Going forward using the `USER kong` as you mentioned is for security reasons -- use the least privileged user possible